pkg/datarepair/repairer/repairer.go

```diff
@@ -29,31 +44,42 @@ type Service struct {
 	repairer  SegmentRepairer
 	limiter   *sync2.Limiter
 	ticker    *time.Ticker
+	pdbServer *pointerdb.Server
```

Comment:
Other servers/endpoints/services/database should not use other servers/endpoints. The Server Endpoint are specifically for communicating outside world, nothing else. The solution here would be to introduce a specific service that allows you to do iteration rather than putting that logic in the server.